python/triton/runtime/build.py

```diff
@@ -40,15 +40,16 @@ def _build(name, src, srcdir, library_dirs, include_dirs, libraries):
     if scheme == 'posix_local':
         scheme = 'posix_prefix'
     py_include_dir = sysconfig.get_paths(scheme=scheme)["include"]
+    cflags = os.environ.get("CFLAGS, "")
     include_dirs = include_dirs + [srcdir, py_include_dir]
-    cc_cmd = [cc, src, "-O3", "-shared", "-fPIC", "-o", so]
+    cc_cmd = [cc, src, "-O3", "-shared", "-fPIC", "-o", so, cflags]
     cc_cmd += [f'-l{lib}' for lib in libraries]
     cc_cmd += [f"-L{dir}" for dir in library_dirs]
     cc_cmd += [f"-I{dir}" for dir in include_dirs]
     ret = subprocess.check_call(cc_cmd)
     if ret == 0:
         return so
-    # fallback on setuptools
+    # fallback on setuptools (dead path due to https://github.com/openai/triton/pull/3311)
```

Comment:
why is that a dead path?

Reply:
if you look at the PR that was reverted, `check_call` throws when the retcode != 0.

that's why I'm surprised that the internal tests failed, that means:
1. the first pass w/o setuptools compilation failed
2. the setuptools compilation failed

Reply:
I would remove this comment, the error I mentioned only happens in some specific cases. Rest looks fine to me.

Reply:
I think there's some kinda of miscommunication? In what scenario can this code be executed?

1. retcode = 0 => return is hit
2. retcode != 0 => exception unwind happens

Reply:
ah right, can you change the comment to something like that then:
```
Currently, this should never happen, since check_call raises an exception when the called process returns a non-zero exit status.
```